app/models/manageiq/providers/vmware/infra_manager/refresh_parser_inventory_object.rb

```diff
@@ -82,7 +82,9 @@ def storage_inv_to_hashes(inv)
           )
         end
 
-        result << result.new_inventory_object(new_result)
+        new_result = result.new_inventory_object(new_result)
```

Comment:
oook

Btw the result_uuids should not be needed, since the result << new_result does the indexing, you can access it even directly with result.data_index

not sure what  result_uids[:storage_id][uid] = new_result does though